pkg/querier/metadata_handler.go

```diff
@@ -61,6 +61,10 @@ func NewMetadataHandler(m MetadataSupplier) http.Handler {
 			metrics[m.Metric] = append(ms, metricMetadata{Type: string(m.Type), Help: m.Help, Unit: m.Unit})
 		}
 
+		if len(metrics) == 0 {
+			w.WriteHeader(http.StatusNoContent)
```

Comment:
Should be a `return` here

Reply:
I had the same thought. 

The closest I can get in prometheus returns a JSON response with just `{"status":"success","data":{}}` immediately after startup with no data volume. Maybe the right thing to do here is just initialize `metrics` to non-nil?

Reply:
Wait, it's already non-nil. 🤔

Reply:
So then remove the `omitempty`? WDYT?

Reply:
Remove the `omitempty` and don't return HTTP 204 sounds correct to me :+1: